doc/manual/source/building.rst

```diff
@@ -0,0 +1,113 @@
+Building From Source
+====================
+
+In order to build dnst from the source, you need to have Rust installed on
+your system. The Rust compiler runs on, and compiles to, a great number of
+platforms, though not all of them are equally supported. The official `Rust
+Platform Support`_ page provides an overview of the various support levels.
+
+While some system distributions include Rust as system packages, Routinator
+relies on a relatively new version of Rust, currently |rustversion| or newer.
+We therefore suggest to use the canonical Rust installation via a tool called
+:program:`rustup`.
+
+Assuming you already have :program:`curl` installed, you can install
+:program:`rustup` and Rust by simply entering:
+
+.. code-block:: text
+
+  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+
+Alternatively, visit the `Rust website
+<https://www.rust-lang.org/tools/install>`_ for other installation methods.
+
+Building and Updating
+---------------------
+
+In Rust, a library or executable program such as dnst is called a
+*crate*. Crates are published on `crates.io
+<https://crates.io/crates/dnst>`_, the Rust package registry. Cargo is
+the Rust package manager. It is a tool that allows Rust packages to declare
+their various dependencies and ensure that you’ll always get a repeatable
+build. 
+
+Cargo fetches and builds dnst’s dependencies into an executable binary
+for your platform. By default you install from crates.io, but you can for
+example also install from a specific Git URL, as explained below.
+
+Installing the latest dnst release from crates.io is as simple as
+running:
+
+.. code-block:: text
+
+  cargo install --locked dnst
```

Comment:
This won't work for now because you'd need `--git`. This is because we can't publish dnst to crates.io until another domain release is made first.